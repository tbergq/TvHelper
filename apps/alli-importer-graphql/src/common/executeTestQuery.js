// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
