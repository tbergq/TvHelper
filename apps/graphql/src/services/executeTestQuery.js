// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
