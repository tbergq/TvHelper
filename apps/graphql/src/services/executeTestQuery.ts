import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
