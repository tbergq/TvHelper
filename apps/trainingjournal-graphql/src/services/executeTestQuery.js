// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

export default function executeTestQuery(query: string, variables: ?{ ... }): Promise<{ ... }> {
  return graphql(schema, query, null, null, variables);
}
