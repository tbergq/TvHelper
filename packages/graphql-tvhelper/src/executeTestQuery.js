// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
