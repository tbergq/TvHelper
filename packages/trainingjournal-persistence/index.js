// @flow strict

import Exercise from './src/dataObjects/Exercise';

// Repositories
export { default as UserRepository } from './src/repositories/UserRepository';
export { default as ExerciseRepository } from './src/repositories/ExerciseRepository';
export { default as ProgramRepository } from './src/repositories/ProgramRepository';

// Connection
export { default as trainingJournalConnection } from './src/connection';

// dataObject

export type ExerciseType = typeof Exercise;
