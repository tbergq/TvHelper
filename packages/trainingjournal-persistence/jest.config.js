// @flow

const path = require('path');

module.exports = {
  rootDir: __dirname,
  testEnvironment: 'node',
  setupFilesAfterEnv: ([path.join(__dirname, 'scripts', 'setupJest.js')] /*: string[] */),
};
