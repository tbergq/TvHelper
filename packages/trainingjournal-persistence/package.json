{
  "name": "@tbergq/trainingjournal-persistence",
  "private": true,
  "version": "0.0.1",
  "main": "index.js",
  "sideEffects": false,
  "dependencies": {
    "mongoose": "^5.9.15"
  },
  "devDependencies": {
    "mockdate": "^3.0.2",
    "mongodb-memory-server": "^6.6.0"
  }
}
