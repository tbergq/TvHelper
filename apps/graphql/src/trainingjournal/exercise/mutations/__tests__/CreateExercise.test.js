// @flow

import { UserRepository, trainingJournalConnection } from '@tbergq/trainingjournal-persistence';

import executeTestQuery from '../../../../services/executeTestQuery';

let user;

beforeEach(async () => {
  user = await UserRepository.createUser({
    username: 'løken',
    password: 'I_am_naive',
    email: 'lol@lol.no',
  });
});
afterEach(async () => {
  await trainingJournalConnection.collection('users').drop();
  await trainingJournalConnection.collection('exercises').drop();
});

it('creates an exercise', async () => {
  expect(
    await executeTestQuery(
      `mutation exerciseMutation($exercise: CreateExerciseInput!) {
        createExercise(exercise: $exercise) {
          exerciseEdge {
            node {
              name
              muscleGroups
            }
          }
        }
      }`,
      { exercise: { name: 'Knebøy', muscleGroups: 'TestLår' } },
      { user: { app: 'trainingjournal', id: user.id } },
    ),
  ).toMatchSnapshot();
});
