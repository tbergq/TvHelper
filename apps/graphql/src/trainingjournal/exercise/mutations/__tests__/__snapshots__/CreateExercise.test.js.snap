// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates an exercise 1`] = `
Object {
  "data": Object {
    "createExercise": Object {
      "exerciseEdge": Object {
        "node": Object {
          "muscleGroups": "TestLår",
          "name": "Knebøy",
        },
      },
    },
  },
}
`;
